src/spatialdata_plot/pl/render.py

```diff
@@ -223,16 +238,44 @@ def _render_shapes(
                 cmap=render_params.cmap_params.cmap,
             )
         )
-        rgba_image = np.transpose(ds_result.to_numpy().base, (0, 1, 2))
-        _cax = ax.imshow(rgba_image, cmap=palette, zorder=render_params.zorder)
-        _cax.set_transform(trans)
-        cax = ax.add_image(_cax)
+
+        # create SpatialImage to get it back to original size
+        rgba_image = np.transpose(ds_result.to_numpy().base, (2, 0, 1))
+        rgba_image = Image2DModel.parse(
+            rgba_image,
+            dims=("c", "y", "x"),
+            transformations={"global": Scale([1, factor, factor], ("c", "y", "x"))},
+        )
+
+        # prepare transformation
+        trans = get_transformation(rgba_image, get_all=True)["global"]
+        affine_trans = trans.to_affine_matrix(input_axes=("x", "y"), output_axes=("x", "y"))
+        trans = mtransforms.Affine2D(matrix=affine_trans)
+        trans_data = trans + ax.transData
+
+        rgba_image = np.transpose(rgba_image.data.compute(), (1, 2, 0))  # type: ignore[attr-defined]
```

Comment:
Here (and in render_shapes), we access the image as numpy array from the SpatialImage. mypy doesn't believe that `compute()` exists...